plugins/inputs/bond/bond.go

```diff
@@ -0,0 +1,172 @@
+package bond
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// default bond directory path
+const (
+	BOND_PATH = "/proc/net/bonding"
+)
+
+type Bond struct {
+	BondPath       string   `toml:"bond_path"`
+	BondInterfaces []string `toml:"bond_interfaces"`
+}
+
+var sampleConfig = `
+  ## Sets bonding directory path
+  ## If not specified, then default is:
+  bond_path = "/proc/net/bonding"
+
+  ## By default, telegraf gather stats for all bond interfaces
+  ## Setting interfaces will restrict the stats to the specified
+  ## bond interfaces.
+  bond_interfaces = ["bond0"]
+`
+
+func (bond *Bond) Description() string {
+	return "Collect bond interface status, slaves statuses and failures count"
+}
+
+func (bond *Bond) SampleConfig() string {
+	return sampleConfig
+}
+
+func (bond *Bond) Gather(acc telegraf.Accumulator) error {
+	// load path, get default value if config value and env variables are empty;
+	// list bond interfaces from bonding directory or gather all interfaces.
+	err := bond.listInterfaces()
+	if err != nil {
+		return err
+	}
+	for _, bondName := range bond.BondInterfaces {
+		file, err := ioutil.ReadFile(bond.BondPath + "/" + bondName)
+		if err != nil {
+			acc.AddError(fmt.Errorf("E! error due inspecting '%s' interface: %v", bondName, err))
+			continue
+		}
+		rawFile := strings.TrimSpace(string(file))
+		err = bond.gatherBondInterface(bondName, rawFile, acc)
+		if err != nil {
+			acc.AddError(fmt.Errorf("E! error due inspecting '%s' interface: %v", bondName, err))
+		}
+	}
+	return nil
+}
+
+func (bond *Bond) gatherBondInterface(bondName string, rawFile string, acc telegraf.Accumulator) error {
+	splitIndex := strings.Index(rawFile, "Slave Interface:")
+	if splitIndex == -1 {
+		splitIndex = len(rawFile)
+	}
+	bondPart := rawFile[:splitIndex]
+	slavePart := rawFile[splitIndex:]
+
+	err := bond.gatherBondPart(bondName, bondPart, acc)
+	if err != nil {
+		return err
+	}
+	err = bond.gatherSlavePart(bondName, slavePart, acc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (bond *Bond) gatherBondPart(bondName string, rawFile string, acc telegraf.Accumulator) error {
+	fields := make(map[string]interface{})
+	tags := map[string]string{
+		"bond": bondName,
+	}
+
+	lines := strings.Split(rawFile, "\n")
```

Comment:
Use a `bufio.Scanner` to avoid EOL encoding issues.

Reply:
done